### PR TITLE
Change AD filter for searchPrincipal

### PR DESCRIPTION
**Pull request by mrajashree**
https://github.com/rancher/rancher/issues/13300

### CODE REVIEW

File: pkg/auth/providers/activedirectory/activedirectory_client.go [hunk 1]
```diff
@@ -323,8 +323,13 @@ func (p *adProvider) searchPrincipals(name, principalType string, config *v3.Act
 }
 
 func (p *adProvider) searchUser(name string, config *v3.ActiveDirectoryConfig, caPool *x509.CertPool) ([]v3.Principal, error) {
-	query := "(&(" + config.UserSearchAttribute + "=*" + name + "*)(" + ObjectClassAttribute + "=" +
-		config.UserObjectClass + "))"
+	srchAttributes := strings.Split(config.UserSearchAttribute, "|")
+	query := "(&(" + ObjectClassAttribute + "=" + config.UserObjectClass + ")"
+	srchAttrs := "(|"
```
Comment (cjellick) [inline pkg/auth/providers/activedirectory/activedirectory_client.go]: What will happen if `srchAttributes` is empty? this search wont function obviously, but i just want to make sure it fails gracefully

Comment (mrajashree) [reply pkg/auth/providers/activedirectory/activedirectory_client.go]: @cjellick It fails gracefully, UI displays 'no options available' in that case

Comment (mrajashree) [reply pkg/auth/providers/activedirectory/activedirectory_client.go]: Should we still make it a compulsory field?

**Comment by cjellick** [comment]
LGTM. types needs updated in vendor

**Comment by cjellick** [comment]
Make sure you let QA know that you also made many of the fields required.

**Pull request merged**